pkg/config/validation.go

```diff
@@ -1,26 +1,37 @@
-package v1beta1
+package config
 
 import (
+	"errors"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/validation"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/kueue/apis/config/v1beta1"
+)
+
+var (
+	errPodOptionsIsNil        = errors.New("value of Config.Integrations.PodOptions is nil")
+	podOptionsPath            = field.NewPath("integrations", "podOptions")
+	errNamespaceSelectorIsNil = errors.New("value of Config.Integrations.PodOptions.PodNamespaceSelector is nil")
+	namespaceSelectorPath     = podOptionsPath.Child("namespaceSelector")
+	errMsgProhibitedNamespace = "namespaces with this label cannot be used for pod integration"
 )
 
-func (c *Configuration) Validate() field.ErrorList {
+func ValidateConfiguration(c v1beta1.Configuration) field.ErrorList {
 	var allErrs field.ErrorList
 
 	// Validate PodNamespaceSelector for the pod framework
 	if slices.Contains(c.Integrations.Frameworks, "pod") {
-		allErrs = append(allErrs, c.validateNamespaceSelector()...)
+		allErrs = append(allErrs, validateNamespaceSelector(c)...)
 	}
 
 	return allErrs
 }
 
-func (c *Configuration) validateNamespaceSelector() field.ErrorList {
+func validateNamespaceSelector(c v1beta1.Configuration) field.ErrorList {
 	var allErrs field.ErrorList
 
 	if c.Integrations.PodOptions == nil {
```

Comment:
use `field.Required`, not InternalError.

Reply:
This field is going to be defaulted if it's not set. That error could only be caused by `kueue` internals. 

I would rather keep it an "InternalError" since `field.Required`could confuse kueue administrator into thinking that they can control that behavior.

Reply:
In kubernetes we follow an approach where defaulting is versioned, while validation is not. So, in theory, we could have a future API version that doesn't set those fields and the user has to manually provide them.

Let's keep the same approach: in other words, don't assume during validation the behavior from defaulting.